### DISCUSSION TITLE
[pro/ent] Store Batch completion time in `Sidekiq::Batch::Status` object

### DISCUSSION THREAD

**Discussion by jai-x** [sidekiq/sidekiq#6135] [Ideas]
After a batch has been completed with no failures, its status is removed from the web UI, but it's data lingers in Redis.

Occasionally, it is useful to know how long the batch took to execute, but this data does not seem to be persisted.

It would be nice if fields like `success_at` and `completed_at` were also stored on the `Status`.

These fields could be populated with timestamp with the same semantics as the `.on(:success, ...)` and `.on(:complete, ...)` callbacks

**Comment by mperham** [comment]
How would you expect to see or get this value? Once a batch is finished, it's considered deleted. Would it be sufficient to publish a Statsd metric for the timing?
